backend/src/services/pki-sync/pki-sync-service.ts

```diff
@@ -245,75 +302,78 @@ export const pkiSyncServiceFactory = ({
       projectId
     });
 
-    ForbiddenError.from(permission).throwUnlessCan(
-      ProjectPermissionPkiSyncActions.Read,
-      subject(ProjectPermissionSub.PkiSyncs, { projectId })
-    );
+    ForbiddenError.from(permission).throwUnlessCan(ProjectPermissionPkiSyncActions.Read, ProjectPermissionSub.PkiSyncs);
 
     const pkiSyncs = await pkiSyncDAL.findByProjectId(projectId);
-    return pkiSyncs;
-  };
-
-  const listPkiSyncsBySubscriberId = async ({ subscriberId }: TListPkiSyncsBySubscriberId) => {
-    const pkiSyncs = await pkiSyncDAL.findBySubscriberId(subscriberId);
-    return pkiSyncs;
+    return pkiSyncs as TPkiSync[];
   };
 
   const findPkiSyncById = async ({ id, projectId }: TFindPkiSyncByIdDTO, actor: OrgServiceActor) => {
+    const pkiSync = await pkiSyncDAL.findById(id);
+    if (!pkiSync)
+      throw new NotFoundError({
+        message: `Could not find PKI Sync with ID "${id}"`
+      });
+
+    if (projectId && pkiSync.projectId !== projectId) {
+      throw new NotFoundError({
+        message: `Could not find PKI Sync with ID "${id}" in project "${projectId}"`
+      });
+    }
```

Comment:
maybe just throw the same error here so that we don't expose to users whether a sync exists in a certain project or not?